internal/lib/utils.go

```diff
@@ -546,3 +547,26 @@ func IsInfraSettingNSScoped(infraSetting, namespace string) bool {
 	}
 	return false
 }
+
+type LockSet struct {
+	locks sync.Map
+}
+
+var lockSet LockSet
+
+func (s *LockSet) Lock(lockName string) {
+	lock, _ := s.locks.LoadOrStore(lockName, &sync.Mutex{})
+	lock.(*sync.Mutex).Lock()
+}
+
+func (s *LockSet) Unlock(lockName string) {
+	if lock, ok := s.locks.Load(lockName); !ok {
+		panic("unlocked an unlock mutex")
```

Comment:
If the lockName(ingress) is not there is the map - shouldn't the unlock operation be assumed successful? 
Why panic in that case?

Reply:
the map will only get populated if the lock for ingress has been called atleast once. this panic should ideally never occur unless there is a code bug. 

we are never deleting the keys in the map, so the keys remain until AKO reboots.